### DISCUSSION TITLE
Is it possible to execute a manual move through javascript/typescript?

### DISCUSSION THREAD

**Discussion by coffeeDev98** [mliebelt/pgn-viewer#401] [Q&A]
Hey @mliebelt ,

I've exposed the PGNViewer object to ```window```. Wanted to know if it was possible to play a manualMove through code.

Any idea on how to do this....?

<img width="362" alt="image" src="https://user-images.githubusercontent.com/63295908/199709483-575bcc36-330c-43aa-8d63-64e03841239c.png">

**Comment by mliebelt** [comment]
See the  ticket #171 with its example. Having the build environment, starting a server locally, going to http://localhost:8080/examples/tickets/ticket171.html, I am able to play (in the console) the moves then.

![image](https://user-images.githubusercontent.com/520039/199745690-1b884ed0-3ea9-4029-a401-0a958ea2af49.png)

So you should be able to use it as well. Please take a look at the example in https://github.com/mliebelt/PgnViewerJS/blob/main/modules/pgn-viewer/examples/tickets/ticket171.html

**Comment by coffeeDev98** [comment]
@mliebelt Yes, I came to this approach as well but there is one issue. For example, when the king is in check the previous turn's king is shown as checked.
In the video below, The manual move is made to move BLACK QUEEN to check the WHITE KING. But, the check highlighting is for the BLACK KING(Previous turn's King, a.k.a, BLACK KING).

Unsure if there are other scenario's where this happens too.

https://user-images.githubusercontent.com/63295908/199759884-e022c6e3-69c3-472f-9496-8546f50a781b.mov

**Reply by mliebelt** [reply]
Ok, I have to check that. But that seems to be a special error and could have to be with timing issues in updating the UI or whatever. The main functionality is there, but it has problems (sometimes). I have added that functionality years ago, due to the demand of someone, and never used it myself.

**Comment by coffeeDev98** [comment]
@mliebelt Can I raise this as an issue for tracking...?

**Reply by mliebelt** [reply]
Of course you can. The problem is, that I have no source code as basis. So you have done something with pgn-viewer as component, adapted it, and then it breaks. At the moment, we have no idea why it breaks (besides some computation inside not done correctly). So it would be way easier if there would be something that reproduces the bug reliable. Else, it is just a documentation of not correct behavior for an unknown (and not visible) reason. I have no problems with open bugs, but I would like to invest my time (which is currently sparse) more in reproducible behavior possible visible to everyone.
